bitswap/server/internal/decision/engine_test.go

```diff
@@ -1433,32 +1433,38 @@ func TestTaggingUseful(t *testing.T) {
 	const peerSampleIntervalHalf = 10 * time.Millisecond
 
 	sampleCh := make(chan struct{})
-	mockClock := clock.NewMock()
+	mockClock := quartz.NewMock(t)
+	trap := mockClock.Trap().NewTicker("scoreWorker")
+	defer trap.Close()
 	me := newTestEngineWithSampling("engine", peerSampleIntervalHalf*2, sampleCh, mockClock)
 	defer me.Engine.Close()
-	mockClock.Add(1 * time.Millisecond)
+	mockClock.Advance(time.Millisecond)
 	friend := peer.ID("friend")
 
 	block := blocks.NewBlock([]byte("foobar"))
 	msg := message.New(false)
 	msg.AddBlock(block)
 
+	// Wait for ticker to be created to ensure clock.AdvenceNext works.
+	ctx := context.Background()
+	trap.MustWait(ctx).MustRelease(ctx)
+
 	for i := 0; i < 3; i++ {
 		if untagged := me.PeerTagger.count(me.Engine.tagUseful); untagged != 0 {
 			t.Fatalf("%d peers should be untagged but weren't", untagged)
 		}
-		mockClock.Add(peerSampleIntervalHalf)
+		mockClock.Advance(peerSampleIntervalHalf)
 		me.Engine.MessageSent(friend, msg)
 
-		mockClock.Add(peerSampleIntervalHalf)
+		mockClock.AdvanceNext()
```

Comment:
```suggestion
		advanced, w := mockClock.AdvanceNext()
		if advanced != peerSampleIntervalHalf {
			t.Fatalf("expected to advance %s, but advanced %s", peerSampleIntervalHalf, advanced)
		}
		w.MustWait(ctx)
```

maybe asserting the advanced interval is not necessary.

Reply:
Probably good to assert the the interval here. However, the MustWait is not needed as the `<-sample` below will do the waiting.